### PR TITLE
Update golangci-lint in Makefile and CI and turn it on in CI

### PR DESCRIPTION
**Pull request by dkropachev**
No functional changes.
Update `golangci-lint`, add it to github workflow, address all linter complaints.

### CODE REVIEW

File: iterx.go [hunk 1]
```diff
@@ -19,17 +19,14 @@ var DefaultUnsafe bool
...
+	fields     [][]int
+	values     []interface{}
 	unsafe     bool
 	structOnly bool
 	applied    bool
-	err        error
-
-	// Cache memory for a rows during iteration in structScan.
-	fields [][]int
-	values []interface{}
```
Comment (sylwiaszunejko) [inline iterx.go:32]: Are we okay with losing the information that this comment gives?

File: batchx.go [hunk 2]
```diff
@@ -4,6 +4,7 @@ import (
 	"github.com/gocql/gocql"
 )
 
+// Batch a batch operations.
```
Comment (sylwiaszunejko) [inline batchx.go:7]: I don't quite get this comment

File: cmd/schemagen/schemagen.go [hunk 3]
```diff
@@ -169,7 +167,7 @@ func existsInSlice(s []string, v string) bool {
 // The second element contains the name of the type.
 //
 //	[["<my_type,", "my_type"] ["my_other_type>", "my_other_type"]]
-var userTypes = regexp.MustCompile(`(?:<|\s)(\w+)(?:>|,)`) // match all types contained in set<X>, list<X>, tuple<A, B> etc.
+var userTypes = regexp.MustCompile(`(?:<|\s)(\w+)[>,]`) // match all types contained in set<X>, list<X>, tuple<A, B> etc.
```
Comment (sylwiaszunejko) [inline cmd/schemagen/schemagen.go:170]: why this change?

Comment (dkropachev) [reply cmd/schemagen/schemagen.go:170]: Regexp simplification, one of the linter does that

File: iterx.go (hunk 1 above)
Comment (dkropachev) [reply iterx.go:32]: thanks for pointing out, fixed.

File: batchx.go (hunk 2 above)
Comment (dkropachev) [reply batchx.go:7]: Added something that makes more sense

**Review by sylwiaszunejko — APPROVED** [review]
